app/assets/javascripts/products.js

```diff
@@ -66,4 +68,29 @@ $(document).ready(function(){
     currentElement.attr("href", wishlist_url);
   }
 
+
+  function setTweet(d,s,id){
+    var message = "Checkout this awesome " + $(".product-name").text();
+    var url = window.location.href;
+    $(".twitter-share-button").attr("data-text", message);
+    $(".twitter-share-button").attr("data-url", url);
+    var js,
+        fjs=d.getElementsByTagName(s)[0],
+        p=/^http:/.test(d.location)?"http":"https";
+
+    if(!d.getElementById(id)){
+      js=d.createElement(s);
+      js.id=id;
+      js.src=p+"://platform.twitter.com/widgets.js";
+      fjs.parentNode.insertBefore(js,fjs);
+    }
+  }
+
```

Comment:
Nice! Could you make the setTweet function more descriptive? Especially with regards to variable names.